src/broker/state_machine.c

```diff
@@ -896,7 +896,7 @@ static void broker_online_cb (flux_future_t *f, void *arg)
     }
 
     if (quorum_reached) {
-        if (s->state != STATE_RUN) {
+        if (s->state == STATE_QUORUM) {
```

Comment:
Commit message suggestion: I had trouble parsing "The quorum-full broker event could be posted in any state but RUN." outside of the Problem statement (i.e. I read it as part of the solution)

Maybe if ", because the quorum-full broker event can be posted in any state but RUN"  was appended to the problem statement it would be more clear? (and the solution would just be "Restrict quorum-full posting to the QUORUM state")

I understand it now, so this should be considered optional.